src/components/id/ImageCarousel/ImageCarousel.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+
+import React, { useMemo, useCallback, memo, useReducer, useEffect } from "react";
+import { IconButton, Box } from "@mui/material";
+import { ArrowBack, ArrowForward } from "@mui/icons-material";
+
+import ImageCarouselProps, { ImageCarouselState } from "@/types/imageCarousel";
+
+import constant from "./ImageCarousel.constants";
+import reducer from "./ImageCarousel.reducer";
+import styles from "./ImageCarousel.styles";
+
+const ImageCarousel: React.FC<ImageCarouselProps> = memo(({ images }) => {
+  const initialState: ImageCarouselState = { images: images, currentImage: 0 };
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      dispatch({ type: "NEXT" });
+    }, constant.intervalTime);
+
+    return () => clearInterval(interval);
+  }, []);
+
+  const ImageComponent = useCallback(
+    () => (
+      <Box
+        component="img"
+        src={images[state.currentImage]}
+        alt={`Slide ${state.currentImage}`}
+        loading="lazy"
+        sx={styles.imgBox}
+      />
+    ),
+    [images, state.currentImage]
+  );
+
+  const dots = useMemo(
+    () =>
+      images.map((_, index) => (
+        <Box
+          key={index}
+          onClick={() => dispatch({ type: "SET", payload: index })}
+          sx={{
+            ...styles.dots,
+            backgroundColor:
+              index === state.currentImage
+                ? constant.dotsColorActive
+                : constant.dotsColorInactive,
+          }}
+        />
+      )),
+    [images, state.currentImage]
+  );
+
```

Comment:
Again, I don't see a need for memoization here, u have 1 state in this component, which means 1 reason to change, memoization would be usable if u had a lot of diffrent states and u don't want additional computation, but here it computes for every state change